### PR TITLE
First draft of data management

### PR DESCRIPTION
**Pull request by jennybc**
**I'm not done yet! Checking in at the 50% mark and will complete this tonight. Vanishing into meetings.**

### CODE REVIEW

**Comment by jennybc** [comment]
OK @gvwilson this is ready for some feedback.

**Comment by gvwilson** [comment]
Thanks @jennybc - @lexnederbragt can you please give this a once-over?

File: index.md [hunk 1]
```diff
@@ -52,8 +52,29 @@ because your past self doesn't answer email.
...
...tream analyses, some of which you have not even dreamed of yet. This is the place to maximize machine and human readability, preferably at the same time. It is neither the place to do vigorous data filtering nor to the time to bring in external information. Approach this initial preparation as internal to the existing dataset and non-destructive, both of the data and its general "shape". Enhance machine readability by converting from proprietary and high-friction formats (e.g., Microsoft Excel or XML) to open and simple formats (e.g., comma delimited plain text). Enhance human readability by replacing inscrutable variable names and artificial data codes with self-explaining alternatives. For example, rename the variable SOME_AWFUL_GENERIC_NAME to SOMETHING_DESCRIPTIVE *(JB: anyone ... got a good example that's real/realistic but will resonate with many?)*, recode the treatment variable from `1` vs. `2` to `untreated` vs. `treated`, and replace artificial codes, such as "-99" for missing data, with proper `NA`s. Both human and machine readability can be enhanced storing especially useful metadata as part of the filename itself, while keeping the filename regular-expression-friendly.
```
Comment (lexnederbragt) [inline index.md:59]: XML is very machine-readable. Just not human-readable...

SOME_AWFUL_GENERIC_NAME --> name1, name2 to firstName, lastName

can be enhanced storing especially --> can be enhanced by storing especially 

Not  sure all readers will be familiar with regular expressions...

File: index.md [hunk 2]
```diff
@@ -52,8 +52,29 @@ because your past self doesn't answer email.
...
+
+*This may still need something about observations and rows and the notion of a key?*
+
+**Choose your friends wisely.** Marshal complementary data in files with the same high standard for openness and simplicity as above. Frequently the raw data does not and indeed cannot fully explain itself. For example, if each well of a microtitre plate is used to study a specific gene knockout, the prepared raw data will have a `well` variable taking on values like "A1" and "G12", but the plate reader cannot possibly know what's in each well. That information will be absent from the primary data file. You must create a supplementary file in order to look this up. Head off future join headaches by taking care to use the same names and codes when variables in two datasets refer to the same thing and will be used for merging or table lookup.
```
Comment (lexnederbragt) [inline index.md:70]: complementary data and supplementary file, could this simply be replaced with the term 'metadata'?

File: index.md [hunk 3]
```diff
@@ -74,7 +95,8 @@ because your past self doesn't answer email.
         transform it for storage with a lossless, well-documented procedure.
     *   Prefer open non-proprietary formats to closed ones (they'll likely last longer)
         *   See [this guide][uiuc-file-formats]
-    *   And don't duplicate contents of stable, long-lived repositories (i.e., don't clone GenBank)
+    *   Consider revoking your own write permission, so it is harder to damage raw data by accident or to hand edit it in a moment of weakness.
```
Comment (lexnederbragt) [inline index.md:98]: Would "Consider restricting file permissions to read-only even for yourself," be more clear?

File: index.md [hunk 4]
```diff
@@ -101,37 +123,11 @@ because your past self doesn't answer email.
...
-> Those plain text files we all love?
-> If you've got people editing or even opening/closing them on different platforms,
-> possibly with locale set to something other than US English,
-> things can get weird.
-> Line endings cycle through \n, \r\n, and \r and "smart quotes" drive you crazy.
->
-> Then you get large and uninformative diffs or the dreaded �.
-
-Regarding metadata, [Elizabeth Wickes][wickes-comment-metadata] writes:
+We propose delimited plain text as an appealing "lowest common denominator" data form, that offers high usability across time, people, operating systems, and analytical environments. We must admit that plain text is, however, no panacea. In particular, when collaborating with others, be aware of and, when possible, standardize on file encoding, line endings, and the elimination of prose-oriented features, such as "smart quotes". Differences between collaborators in these pesky details can cause the "diffs" between file versions to be unnecessarily large and, therefore, to be substantially less informative.
```
Comment (lexnederbragt) [inline index.md:128]: Not sure all readers will know what you mean with 'diff'

**Comment by lexnederbragt** [comment]
Nice text! We may have to discuss whether we remove all bullet points for all sections - and if we do there is a bit of rewriting to do for this part...

**Comment by jennybc** [comment]
@lexnederbragt Thanks! Yes I have many question marks about the bulletin and also tone/voice/length. But it's a start. Thanks for your suggestions, I will act on them.

File: index.md (hunk 1 above)
Comment (gvwilson) [inline index.md:59]: Break the specific recommendations out into bullet points.

File: index.md [hunk 5]
```diff
@@ -52,8 +52,29 @@ because your past self doesn't answer email.
...
+
+  1. Columns that contain more than one variable's worth of information. For example, the inclusion of units is problematic, e.g., "3.4 kg". The presence of "kg" will cause most analytical environments to read this in as character data, whereas you'll probably want to do numeric things with it, like take averages and use in plots. This should be split into two variables, the mass "3.4" and the units "kg" (or the units should be recorded in the variable name and/or metadata). When in doubt, try to make each variable correspond to an atomic, imminently usable piece of information.
+  2. Multiple columns that -- only when taken together -- contain one variable's worth of information. This is characteristic of data that has been laid out for human eyeballs or for manual data entry. For example, there might be one row per field site and then columns for measurements made at each of several time points. For data entry and inspection, it is convenient to store this in a "short and wide" form, but for most analyses it will be advantageous to gather these columns into a variable of measurements, accompanied by a companion variable indicating the time point.
```
Comment (gvwilson) [inline index.md:64]: Add two example tables, one showing human-oriented "short and wide", the other what we want.

File: index.md [hunk 6]
```diff
@@ -52,8 +52,29 @@ because your past self doesn't answer email.
...
+
+The goal of all this variable splitting, combining, spreading, and gathering is to create so-called "tidy data", which can be a powerful accelerator for analysis ([Tidy Data][wickham-tidy], [Nine simple ways][white-simple-reuse]). Other considerations: this is a good place to make sure your data will play nicely with your analytical environment and plans. For example, reformat a date-time to match one of those recognized automatically or standardize the casing on a character variable. This pass through the data likely does not change the amount of data, but may dramatically alter its form.
+
+*This may still need something about observations and rows and the notion of a key?*
```
Comment (gvwilson) [inline index.md:68]: Yes please and yes please.

File: index.md (hunk 3 above)
Comment (gvwilson) [inline index.md:98]: Haven't run into this one before, but I like it.

File: index.md [hunk 7]
```diff
@@ -74,7 +95,8 @@ because your past self doesn't answer email.
         transform it for storage with a lossless, well-documented procedure.
     *   Prefer open non-proprietary formats to closed ones (they'll likely last longer)
         *   See [this guide][uiuc-file-formats]
-    *   And don't duplicate contents of stable, long-lived repositories (i.e., don't clone GenBank)
+    *   Consider revoking your own write permission, so it is harder to damage raw data by accident or to hand edit it in a moment of weakness.
+    *   Don't duplicate contents of stable, long-lived repositories (i.e., don't clone GenBank)
```
Comment (gvwilson) [inline index.md:99]: I like this too - where can it be added?

File: index.md [hunk 8]
```diff
@@ -101,37 +123,11 @@ because your past self doesn't answer email.
...
-> They are to describe the data set as a unit,
-> e.g. author, funder, relevant papers, etc.
->
-> Formally structured metadata is often a valueless effort if the data set will be stored independently
-> and not somewhere in a formal repository...
-> Beautifully filled out metadata files are for ingestion into a repository and/or directory.
-> If the audience is humans, write it for humans.
-> If the audience includes metadata harvesters,
-> fill out the formal metadata and do a README for the humans.
+[Elizabeth Wickes][wickes-comment-metadata] provides a useful classification for metadata, with implications for it how it should be represented. She notes it is easy to conflate metadata about the dataset as a whole with metadata about the content, e.g., individual columns. Most metadata schemas are aimed at the former, i.e. they detail the author, funder, related publications, etc. When considering how to store metadata, consider the intended audience. Is it humans? Write a README. Is it machines, such as metadata harvesters and formal repostiories? Create an impeccably formated metadata file.
```
Comment (gvwilson) [inline index.md:130]: Readers of this doc are unlikely to know the difference between peccably and impeccably formatted metadata without an example.

**Comment by gvwilson** [comment]
ETA for merge?

**Pull request merged**